File: src/main/java/com/diskria/advancementsfullscreen/injection/extension/AdvancementsScreenExtension.java
package com.diskria.advancementsfullscreen.injection.extension;

import net.fabricmc.api.EnvType;
import net.fabricmc.api.Environment;
import net.minecraft.client.MinecraftClient;

@Environment(EnvType.CLIENT)
public interface AdvancementsScreenExtension {
    int advancementsfullscreen$getWindowWidth(boolean isWithBorder);

    int advancementsfullscreen$getWindowHeight(boolean isWithBorder);

    int advancementsfullscreen$getWindowHorizontalMargin();

    int advancementsfullscreen$getWindowVerticalMargin();

    void advancementsfullscreen$resize(MinecraftClient client, int width, int height);
}

File: src/client/kotlin/io/github/diskria/advancements/fullscreen/extensions/AdvancementsScreenExtension.kt
package io.github.diskria.advancements.fullscreen.extensions

import net.fabricmc.api.EnvType
import net.fabricmc.api.Environment
import net.minecraft.client.MinecraftClient

@Environment(EnvType.CLIENT)
interface AdvancementsScreenExtension {
    fun advancements_fullscreen_getWindowWidth(isWithBorder: Boolean): Int
    fun advancements_fullscreen_getWindowHeight(isWithBorder: Boolean): Int
    fun advancements_fullscreen_getWindowHorizontalMargin(): Int
    fun advancements_fullscreen_getWindowVerticalMargin(): Int
    fun advancements_fullscreen_resize(client: MinecraftClient, width: Int, height: Int)
}
